### DISCUSSION TITLE
Installing 0.8.0 on Manjaro 24.1 (rolling release) some observations about Accu-Chek meters

### DISCUSSION THREAD

**Discussion by uncledulgaria** [andyrozman/ggc#6] [General]
I've now got a working install, on the way I found a couple of issues I'd like to share, either to help others or flag up some potential issues.
Initially I had to find apache-maven-3.6.3 as a tar.gz file on their archive and then install the binaries as per the included README file.

Firstly "Build".
Using "git clone git@github.com:andyrozman/ggc.git --recursive" - it failed to download the atech-tools element due to a problem with the fingerprint.
I got a yes/no/fingerprint request, as I don't have an ssh key, the permission was not granted and that part of the clone failed.  Why does the atech-tools need ssh key but the main code not need one ?

The workaround was to download the atech-tools zip file and manually copy the contents into the atech-tools folder.
Running "mvn clean install" meant I got a fully successful build of ggc.

Secondly, I ran "java -jar ggc-desktop-0.8.0-SNAPSHOT-jar-with-dependencies.jar" from the ggc-desktop-app/ggc-desktop/target folder.

On first run, having failed to find the data folder, it stared searching what seemed to be a series of random folders which seem to have data: at the end of the path. - is this correct behaviour ?

17:24:32,405  INFO [GGC:93] - =============== Starting GGC (0.8.0 - 24.11.2024 17:13) ===============
17:24:32,409  INFO [UserDataDirectoryV2Validator:60] - Directory UserDataDirectory: /home/doug/.GNU_Gluco_Control found.
17:24:32,410  INFO [UserDataDirectoryV2Validator:69] - check If All Required Files Exist
17:24:32,411  WARN [UserDataDirectoryV2Validator:83] - File /home/doug/.GNU_Gluco_Control/db/ggc_db.mv.db exists: false
17:24:32,411  WARN [UserDataDirectory:231] - Installation doesn't seem to be valid. Running creation or migration of data directories
17:24:34,004  WARN [UserDataDirectoryV2CreatorOrMigration:149] - determineIfThisIsCorrectDataDirectory: /home/doug/.exact-audio-copy/dosdevices/z:/usr/lib64/qt6/mkspecs/features/data: false
17:24:34,409  WARN [UserDataDirectoryV2CreatorOrMigration:149] - determineIfThisIsCorrectDataDirectory: /home/doug/.exact-audio-copy/dosdevices/z:/usr/lib64/libreoffice/sdk/examples/DevelopersGuide/UCB/data: false
This continued for several seconds, so I did Ctrl-C to end the java command, tried to work out why it was looking in strange folders, then ran the command again.

17:26:40,491  INFO [GGC:93] - =============== Starting GGC (0.8.0 - 24.11.2024 17:13) ===============
17:26:40,497  INFO [UserDataDirectoryV2Validator:60] - Directory UserDataDirectory: /home/doug/.GNU_Gluco_Control found.
17:26:40,497  INFO [UserDataDirectoryV2Validator:69] - check If All Required Files Exist
17:26:40,498  WARN [UserDataDirectoryV2Validator:83] - File /home/doug/.GNU_Gluco_Control/db/ggc_db.mv.db exists: false
17:26:40,499  WARN [UserDataDirectory:231] - Installation doesn't seem to be valid. Running creation or migration of data directories
17:26:40,716  WARN [UserDataDirectoryV2CreatorOrMigration:149] - determineIfThisIsCorrectDataDirectory: /home/doug/ggc/ggc.try1/ggc-desktop-app/ggc-desktop/data: false
17:26:40,728  INFO [UserDataDirectoryV2CreatorOrMigration:105] - Trying to installNewDataDirectory
17:26:40,739  INFO [InstallMigrationHandler:64] - Process Install
17:26:40,742  INFO [InstallMigrationHandler:196] - Unzip resource file '/install/db_7_v2.zip' to target the ATdirectory.
17:26:40,966  INFO [InstallMigrationHandler:196] - Unzip resource file '/install/data_root.zip' to target directory.
17:26:40,971  INFO [InstallMigrationHandler:196] - Unzip resource file '/install/skinlf_themes.zip' to target directory.
17:26:41,135  INFO [InstallMigrationHandler:196] - Unzip resource file '/install/nutrition.zip' to target directory.
17:26:42,012  INFO [InstallMigrationHandler:251] - Make directory tools.
17:26:42,045  INFO [InstallMigrationHandler:251] - Make directory export.
17:26:42,047  INFO [InstallMigrationHandler:251] - Make directory print.
17:26:42,294  INFO [HibernateConfiguration:180] - Your DB version identified by HBM.xml resource files (sha1=17f84910f226c9d9bf6c8218fc824c6d2a01d05b, db_version=7).
17:26:42,306  INFO [HibernateConfiguration:443] - Loading Db Configuration #0
17:26:42,307  INFO [GGCDbConfig:158] - GGCDb: Loading Db Configuration #0: Internal Db (H2)
17:26:42,999  INFO [DataAccess:1512] - Start internal Web Server
17:26:43,042  INFO [Server:264] - Pygmy Server (v0.4.3) running.
17:26:43,067  WARN [DbCheckReport:148] - Db Report file not found !
17:26:43,118 DEBUG [RedirectScreen:170] - SkinLf file: /home/doug/.GNU_Gluco_Control/skinlf_themes/modernthemepack_ggc.zip, exists: true
^C 17:26:43,679  INFO [Server:388] - Pygmy Server (v0.4.3) shutdown complete.

I did Ctrl-C again, to check the messages, just as the app window opened, this time it looked Ok.

I've explored the features of the app, but when as I tried to connect my SmartPix unit, it failed, and so checked the help, then I discovered that you haven't got the SmartPix v2 to work on linux, which is parallel to my experience with running the SmartPix app under wine.  The app works and manipulates the database etc., but won't communicate so I have to use a windows install to read my device.

I hope to come back later and explain why a type 2 diabetic is testing at all, and trying FOSS solutions instead of using the proprietary apps that are available.

Regards
Doug

**Comment by andyrozman** [comment]
Smartpix should be working correctly on Linux... It was tested and implemented on Linux. Try connecting Smartpix before you start app and test again. Or maybe you have higher version of Smartpix... I am affraid that I can't have have a way to test it at the moment. I would need to get myself a Roche meter first. I will see what can be done... 

But Smartpix should be supported. I used it a lot in the past, but I haven't tested it with new version. 

Do 2 tests:
1. Attach Smartpix and then start application and try to read (see if there is anything in the logs)
2. Don't have Smartpix attached and start the application and connect Smartpix then and try to read 
 
(you can select any Roche  meter, they all work the same).

And you don't have to explain anything... Testing is good for both types and most of software doesn't work on Linux. I haven't touched this app much before 1 year ago when I managed to move it to github, because people just didn't use it anymore (or at least they didn't tell me they did)...

Where did you find information that SmartPix 2.x doesn't work on Linux?

**Reply by uncledulgaria** [reply]
Your own help popup.
![Screenshot from 2024-11-25 10-42-49](https://github.com/user-attachments/assets/795365c2-bb3b-457c-89b8-3f8782caed19)
I'll admit I didn't manage to get the mount aspect of the configuration correct before making the comment, but I have spent many hours researching wine and USB connectivity, trying to resolve the problem of making my SmartPix 2 device work on Linux.  Instead, I boot into windows 10, run the software to do the data download from the meter to a shared diaries folder, then reboot back into linux to do any further manipulation of the data.

**Comment by andyrozman** [comment]
Would it be possible for you to upgrade to SmartPix 3? I mean SmartPix 3 has been around for over 16 years now...

**Reply by uncledulgaria** [reply]
Sadly, my doctor disagrees with me (and you) about testing.  In England, if they follow the NICE guidelines, type 2's don't need to test unless they are on medication that can cause hypos.  I'm now on metformin, as well as a low carb diet, which I have managed by literally "eating to my meter".  My HbA1c is 35.  Despite my protests, they stopped my supply of test strips and lancets at the end of September, I'm now having to buy my own and the test strips for the Accu-Chek Performa Nano I was given are £20 per 50 at retail.  I've just bought the Instant as the test strips are around £9 per 50. I had thought that the Instant worked with the SmartPix software, but it doesn't, as stated earlier.  Hence for a request for the Instant to be added.  Is there any way I can help  ?  I'm not a coder, but very happy with the hardware/software interface, and familiar with command line stuff relating to Linux.

**Reply by andyrozman** [reply]
Can you go to your SmartPix and select START.HTM in browser, it should show you the version... Problem might actually be with your version of Smartpix... Since Instant is very new device, it might not work with older versions of smartpix... I know that I had to upgrade my SmartPix when I started using the Combo pump (which was also ages ago)...

**Reply by andyrozman** [reply]
Question is if firmware upgrade is still possible... I was looking at this all the downloads for Smartpix and I can't find the one that actually changes firmware... And it seems that new browser can't even open smartpix directly anymore... If you got your Smartpix in last few years, you should have version 3 already...

**Reply by andyrozman** [reply]
So if you go to <drive>/START.HTM and open in browser you should get something like this; 
![image](https://github.com/user-attachments/assets/4a05e17b-ce4e-4019-9a77-c556ea60ec49)

What is your version shown there?

**Comment by andyrozman** [comment]
I have been using GGC on Linux from the start... So it should work... You should have latest SmartPix installed (which I think its 3.0.5). You can check it on SMARTPIX device in folder misc there is file S* and it shows your serial number and Firmware version. You can easily upgrade by downloading firmware upgrade package... 

Here is download for upgrade: https://www.accu-chek.ch/produkte/smartpix  (it seems actual version is 3.2.5)

**Reply by uncledulgaria** [reply]
On my SmartPix device, folder MISC contains  file V3022552.LOG with the following contents :-
SerialNo:   GV03022552
Eeprom:     1.00.00
Hardware:   4
Bootware:   1.00.00
Firmware:   3.04.01
Resources:  3.02.00
DMR Rev.:   0

I have been using 3.2.5 on both Win10 and Manjaro/wine since I set my current system up in November last year, it was released in 2022.

Today I realised how I was supposed to connect the Accu-Chek Instant meter that uses a USB lead instead of IR.
The lead has to connect to the USB socket on the underside of the SmartPix device in order that the SmartPix software can connect to it.  If I connect the Instant meter directly to the PCs USB port, it isn't recognized.
I have now successfully read data from my new Instant into the SmartPix software, so I'm a lot happier, as it means I don't have to use the MySugr app and keep my data in Roche's cloud, I can keep on using the system I have been running for the last 12 months without any changes.  

It does of course mean that I have to reboot into Win10 to read the meter initially, but today I found that SmartPix software on Manjaro/wine can talk to the SmartPix device and the two meters under some very particular circumstances, but it needs more research to work out what is going on.

**Reply by andyrozman** [reply]
Before when everything worked you could read all the data directly to GGC with Smartpix, but seems like SmartPix doesn't work "correctly" anymore... At least not with my implementation... I can still issue the commands, but it seems like status (in STATUS.TXT file) is no longer changing... I will have to restest it and see if I have broken something in my implementation or if SmartPix doesn't work like it should...

There should be a way to import data into GGC. Problem is that with lot of changes I had to do from 0.7 to 0.8, I am not sure if that is working still... I had to do big changes to make project runable from single jar file and to store settings in "User data directry", and after change I just didn't have time to test everything... My previous co-worker did most of the testing and finding of problems and now that he is gone from project this is left to me... And as you can see project is not used that much, since most of people moved to the cloud and only people using it are those on Linux and Mac, because most of diabetes software works only on Windows (or now on Android)...

**Reply by uncledulgaria** [reply]
I'm happy to try and assist with some testing, I have three meters available to me :-

1. A.Menarini GlucoMen Areo, with USB lead and their Windows software to verify connectivity.
2. SmartPix Model 02 device.
3. Accu-Chek Performa Nano (I/R)
4. Accu-Chek Instant (USB port & Bluetooth)

My reasons for wanting to help are that I have data, usually in CSV formats, from two meters and several proprietary applications or cloud storage systems.  I'd like to be able to review all of it under the one system in order to make some sense for the development of a strategy for management of my diabetes, that is now being made much more difficult due to Long COVID.  I'm also on a carb restricted diet.

**Reply by andyrozman** [reply]
Any testing is welcome... Although until I look at SmartPix with my devices (which will happen in 2 weeks) it doesn't make sense to test those... But you could theoretically test Glucomen meter... 
You can just go around application and click different kind of options to check if everything works... If you run it from command line you should be able to see logs and if something goes wrong... If you find something write it down and we can see and maybe open tickets, just to keep an eye on things... 

I just wondered... My SmartPix doesn't have USB port and its 2017 model... Is your newer?  

From what kind of software do you have exports? Maybe I could whip up the way to do imports... I had plans to do another plugin (connect), which would be used exactly for things like this... imports from other software packages... Problem is that it never happened... If you want you could send me your exports and I could see if I can easily import them. If you have any export information (from software you exported) that would also be a nice thing...

**Reply by uncledulgaria** [reply]
1.  Done a quick test with the glucomen areo, generates the same error over and over again.
22:08:40,757 DEBUG [SelectorAbstractDialog:592] - SelectorAbstract 0.2
22:09:30,108 DEBUG [CommunicationPortSelector:93] - Selected Discovery Agent: SerialDiscoveryAgent
check_group_uucp(): error testing lock file creation Error details:Permission deniedcheck_lock_status: No permission to create lock file.
 FAILED TO OPEN: No such file or directory
The error started as soon as I clicked on the serial port selection.
I've checked and I am a member of the uucp group.
No choices appear in the dialogue box.
Just a word about Manjaro, it's based on Arch Linux, (so uses the same pacman package management tool and shares some parts of their repos), systemd, and I'm on kernel 6.6.54 at the moment.

2. I suspect my SmartPix device is older, all the folders in the storage all dated 202-01-01, except the SMARTPIX.SW folder is dated 1997-01-01, but the files inside it are dated 2017-01-01 (perhaps it was updated as you described).  I bought it off Ebay for £30 last year.  Here is a link to one like it. https://www.ebay.co.uk/itm/186804830012  
N.B I've also got one of those RealTyme devices as well.  That worked with the Diasend (now Glooko) upload software as well as Accu-Chek 360 software, I only moved to SmartPix when we were forced to move from Diasend (Sweden) to Gooko (USA).

3. Software - the Menarini software that worked with the Glucomen Areo (free download), Diasend/Glooko (cloud), and the Accu-Chek SmartPix Software.
I have csv exports for the first two, and the Diary XML format files from the SmartPix.  I also have database backups from the Menarini software.
The problem with the Diasend CSV files is that they don't seem to have exported the pre/post meal tags, just the date/time and BG levels.
I'll try and put a properly documented package together for you, but life is getting rather complicated at the moment, I have surgery due next week, plus several appointments just after that.  This has been a most welcome distraction from all of that.

**Reply by andyrozman** [reply]
Mhm... Your SmartPix is different model then mine (I have probably Model 01)... I will see if I can get that one that you sent me link for... 

I think that I needed special cable for Glucomen. In the past there were special cables available (for all meters) that actually had special hardware in them, they had standard connectors, but that special hardware. Which might be a reason why this doesn't work... 

Don't worry, there is no hurry to prepare that package, if you are very busy... But I imagine distractions are welcome, especially if you are stressed because of surgery...

If you have time you can test everything else... Software has a lot of "secret" functinalities that are nowhere commented...

**Reply by uncledulgaria** [reply]
Correction, that 202-01-01 date should be 2012-01-01.

**Reply by uncledulgaria** [reply]
I think I'm beginning to understand why I couldn't find start.html or status.txt, there seems to have been a significant difference between these devices, and why you are seeing different software versions on the SmartPix device to me.
My SmartPix device has the following characteristics :-
When plugged into USB when running linux, the SmartPix device appears as a USB drive labeled SMART_PIX2. 
 
If I run lsusb I see :-
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 002: ID 13ee:0001 MosArt Optical Mouse
Bus 002 Device 014: ID 173a:083c Roche Accu-Chek Smart Pix Hub
Bus 002 Device 015: ID 173a:083b Roche Accu-Chek Smart Pix

These relate to the two external USB ports on the thinkpad.

and lsusb --tree results for Bus 002

Bus 002.Port 001: Dev 001, Class=root_hub, Driver=xhci_hcd/9p, 480M
    |__ Port 001: Dev 002, If 0, Class=Human Interface Device, Driver=usbhid, 1.5M
    |__ Port 002: Dev 014, If 0, Class=Hub, Driver=hub/2p, 12M
        |__ Port 002: Dev 015, If 0, Class=Mass Storage, Driver=usb-storage, 12M
        |__ Port 002: Dev 015, If 1, Class=Human Interface Device, Driver=usbhid, 12M

i.e. it appears as a USB hub, an HID device and USB storage. 
The USB storage contains just a software installation package for Windows of the SmartPix software (version 3.0.0), no firmware or other files.  The windows install defaults to "C:\Users\myname\Documents\Accu-Chek Smart Pix Software".  In this folder is both the main executable, a bunch of .dlls and the Diaries folder containing the downloaded meter data.  It is **not** installed in Program files, so I suspect it can be run as a portable app if needed.

What you were calling version 1.x 2.x and 3.x in that help that I screenshotted earlier in this thread is related to the embedded code, whereas I was thinking it related to the SmartPix device physical hardware.  Mine says "Model 02", and I can see on ebay there is now a V3, that is rectangular. https://www.ebay.co.uk/itm/266843545941.

I'm beginning to think that the HID connection on the device is to some sort of USB to serial uart, either through a basic microUSB cable to the Instant, or via I/R to the Performa Nano.

If your device proves to be a Model 01, then there may have been a significant development at Roche in their interface to the 02.
However, the good news is that I can trick my Model 02 to interface with the SmartPix software running under Wine on Linux, so it may be possible for me to sniff the communications between them.  I've already managed to get Wireshark to track some USB related data (URB) on my thinkpad, but I need to some more reading to actually understand what I'm looking at.

**Reply by andyrozman** [reply]
Discovering how it works will be nightmare... If I didn't have help from my "inside" man then I could have never implemented it... 



This is how mine looks https://www.ebay.ie/itm/275420780086?_skw=smart+pix&itmmeta=01JDPV3XPZ862ZGMM6HCVSC6GR&hash=item40205b9636:g:D1kAAOSwYYtgdxyX&itmprp=enc%3AAQAJAAAA4HoV3kP08IDx%2BKZ9MfhVJKnHBUddGXkZdHonKkQlZekAQYdbCT%2FBW24n%2BdDLbhVwo%2FWjpYcn%2BejWCmd0EI2fFPlEQLf3fe4Y8CGo6uDmGxw6cdRcIc1lCTXwlO5pGR2V%2BDN%2B81IRSk7OD%2FqyaDjkEWF7r55UGn9JJ4zEN1H2WvzfUSjOoEfrQ8qwyg7v2IAn0tlKhnGdOWh%2FtJXhOClmq8S9NgQRo6W2dhAkQWCA2aFmdmBWicPgPrT82PqSDUbNJ1DlJk3jEmFRtgu0nbbmWoeBW9d8b8KQTqXbg1PG7PWm%7Ctkp%3ABk9SR8rbj9vtZA

All silvery and very shiny...

**Reply by uncledulgaria** [reply]
I've just done what I should have done before - read the manual.
It explains in great detail the way to connect of the two types of Smart Pix devices when using the Smart Pix Windows software.
It seems that your device is a Model 1, and mine is a Model 2.  The same Windows application works for both, but with different connectivity for non  I/R devices.  Here is an edited version of the latest manual that was contained within the Windows install (lots of pages deleted) that explains the difference between Models 1 and 2 connnections.  
[Accu-Chek Smart Pix Manual-EN-3.2.5(02)-reduced.pdf](https://github.com/user-attachments/files/17936977/Accu-Chek.Smart.Pix.Manual-EN-3.2.5.02.-reduced.pdf)
Earlier versions of the manuals (2016) indicate two interfaces to the devices.  The Basic one uses a browser and looks like that screenshot you sent that used START.HTML, and the Advanced one is the Windows desktop version which I use that is described in the attached manual.
I don't know where this takes us, but until you can hook your Model 1 up to the latest GGC code I don't think we can do anything with Accu-Chek.

I will however look at the Menarini device more closely, and feed back any issues on a separate thread.

**Reply by andyrozman** [reply]
He he... But that is actually just User manual, it doesn't give details what commands needs to be issued to device and in which way and what response to expect. 

Old model worked that you issued speciafic command (in form of creating file with specific data) and then SP would run and generate special XML file on the drive, which you could then read. Until I get my hands on this "Model 2" device I won't know if it works the same way or is this totally different "beast"... And then I also to need to determine what is happening with "Model 1" why it stopped working.

**Reply by uncledulgaria** [reply]
Where exactly did that command file have to be put in order for it to be processed, what was it's name  ?

**Reply by andyrozman** [reply]
The old SP, had sort of Read-Only file system, you could see the files, but you couldn't add them or delete them. So when you copied file (with any name) to it and if it had correct content it would execute command specified in file:

`SL42-B RemoteControl File
Command=EraseReport`

This would be command to erase report, which after reading would reside on:
/SMARTPIX/REPORT/XML/I<serial>.XML
this would be report for Insulin pump and if it looked 
/SMARTPIX/REPORT/XML/G<serial>.XML
it woul be report for Glucose Meter.

**Comment by andyrozman** [comment]
Ok. So I don't know what happened, but it seems that SmartPix doesn't work like it worked before... At least not the version I have... I will have to check and see if I can rewrite this... Really weird... I will have access to my old meters in about 2 weeks, so I will check then... But at least I can see that reading pump with Smartpix doesn't work... I have different version of Smartpix at my parents place and I have to see if that one works better...

**Reply by uncledulgaria** [reply]
Maybe there is a clue in the strange thing I found yesterday.  Please read carefully, it's a bit complicated, I've deliberately used the terms "SmartPix device" and "SmartPix software" throughout to make them distinct.

In the past, I have failed to get the Accu-Check SmartPix (v3.2.5) Windows software that I installed under Wine to connect to the SmartPix device, it just times out.  I have been through all the items relating to USB connectivity that I could find in the Wine help and wider web.  Dual booting was the only solution.
The same laptop running Windows 10 with the same SmartPix software and device works perfectly, so it's definitely not a hardware issue.

Yesterday, once I had grasped the need to keep the SmartPix device in circuit in order to connect the Instant meter, I had done a successful connection to the Instant when in Windows.  The blue speckled light on the SmartPix device lights up when reading is ready or taking place for both the I/R and USB meters.  Before yesterday, I've never had the blue speckled light lit on Linux.

I then restarted the thinkpad into linux, without powering down, however the **USB lead was unplugged** to prevent the thinkpad from trying to boot off the USB mass storage part of the SmartPix device.
On logging into Linux, I first opened the SmartPix app, then plugged the SmartPix device back into the same USB port and did a successful download from both meters **in Linux**, something that has never happened before.

This process is repeatable, and I can close the app and re-open as many times as I like and there is still a connection to the SmartPix device.

However, once the device has been disconnected, the app cannot reconnect without returning to Windows 10 and doing the same thing over again.

What I haven't yet checked out is whether the USB device numbering changes on reconnection, I suspect it does, but I think the issues might be related to data rate or handshake rather than port numbering.
My thinking is that speed or handshake is initially being set by Win10, is **not** unset during the reboot and initial connection, but reset on re-connection of the device.  Somewhere along the way, SmartPix software under wine is unable to set the correct parameters.

Maybe this aspect is where you might need to start your investigations.  Perhaps the default USB data speed in Linux is now faster than the old SmartPix device can handle, in seven years a lot can change.

**Reply by andyrozman** [reply]
Something else is wrong... I never had problem running SP on my Linux.. We have implementation where we issue special command to SP and when thoose commands are executed STATUS.TXT is supposed to be changing (with status of what is happening. For some reason this doesn't work like this anymore (they might have broken this and not noticed, since they might have stopped doing it this way... Years ago I had person in Roche, which I could contact and I had inside view how that works... And it worked great... But its been over 15 years since I implemented SP support, so it seems that in meantime they had several upgrades and have broken this functionality. So I need to retest it with device that I implemented it with, which is not where I currently live...

**Reply by andyrozman** [reply]
Ok. So bad news/bad news. I have tested SmartPix with older version and it seems that some of my changes in past have broken all imports. So older version works "better", but it seems SmartPix is still not working. 

So I need to fix import code in GGC 0.8 (at least for SmartPix) and then I can look at how to fix SmartPix itself. So when we start import we have sort of PreInit part, which turns of AUTO SCAN and then removes current file. This is the part that wasn't working in new version, so after I add this functionality back (I currently have 2 version of "drivers" running in application V1 which is old and V2 which is new one, and to which I am moving all old device implementations, but seems that when I create V2, I missed out preInit functionality (at the moment SmartPix (and devices using it) are only devices that uses that). 

So after that is fixed  then I can retest SmartPix and find solution to why it doesn't work (and how to make it work again)... Since this doesn't work on old version either (where it worked before), I am assuming that something within SmartPix Model 1 has changed, which is the reason why it stopped working.
